cmd/krew/cmd/list.go

```diff
@@ -56,17 +134,42 @@ Remarks:
 				return nil
 			}
 
-			// print table
-			var rows [][]string
-			for p, version := range plugins {
-				rows = append(rows, []string{p, version})
+			if *output.OutputFormat == "wide" || (*output.OutputFormat == "" && isTerminal(os.Stdout)) {
+				// print table
+				var rows [][]string
+				for p, version := range plugins {
+					rows = append(rows, []string{p, version})
+				}
+				rows = sortByFirstColumn(rows)
+				return printTable(os.Stdout, []string{"PLUGIN", "VERSION"}, rows)
 			}
-			rows = sortByFirstColumn(rows)
-			return printTable(os.Stdout, []string{"PLUGIN", "VERSION"}, rows)
+
+			if slice.ContainsString(output.AllowedFormats(), *output.OutputFormat, nil) {
```

Comment:
This `if` seems to repeat the logic from `ToPrinter`. It is safe to call `ToPrinter` with arbitrary OutpuFormat, because this will detect a reasonable error. However, you should move the `ToPrinter` call upfront to catch this error early. Like so:
```patch
diff --git a/cmd/krew/cmd/list.go b/cmd/krew/cmd/list.go
index c478981..e6e25c1 100644
--- a/cmd/krew/cmd/list.go
+++ b/cmd/krew/cmd/list.go
@@ -28,8 +28,6 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/cli-runtime/pkg/genericclioptions"
 	"k8s.io/cli-runtime/pkg/printers"
-	"k8s.io/kubectl/pkg/util/slice"
-
 	"sigs.k8s.io/krew/pkg/constants"
 	"sigs.k8s.io/krew/pkg/installation"
 )
@@ -144,25 +142,21 @@ Remarks:
 				return printTable(os.Stdout, []string{"PLUGIN", "VERSION"}, rows)
 			}
 
-			if slice.ContainsString(output.AllowedFormats(), *output.OutputFormat, nil) {
-				objs := make(Entries, 0, len(plugins))
-				for plugin, version := range plugins {
-					obj := Entry{plugin, version}
-					objs = append(objs, obj)
-				}
-				objs = sortByName(objs)
-				p, err := output.ToPrinter()
-				if err != nil {
-					return err
-				}
-				err = p.PrintObj(objs, os.Stdout)
-				if err != nil {
-					return err
-				}
-				return nil
+			p, err := output.ToPrinter()
+			if err != nil {
+				return err
 			}
-
-			return errors.New("unsupported output format specified")
+			objs := make(Entries, 0, len(plugins))
+			for plugin, version := range plugins {
+				obj := Entry{plugin, version}
+				objs = append(objs, obj)
+			}
+			objs = sortByName(objs)
+			err = p.PrintObj(objs, os.Stdout)
+			if err != nil {
+				return err
+			}
+			return nil
 		},
 		PreRunE: checkIndex,
 	}
```

Reply:
I made it this way to assist in readability; name, wide and json/yaml are all distinctly handled with other options throwing an error.

Also, I realize now this is why AllowedFormats() only lists yaml/json support.

Reply:
I'll move the ToPrinter up though.